File: dotnet/Allmark/Render/ConsoleLinkRenderer.cs
namespace Allmark.Render;

using Allmark.Types;

public static class ConsoleLinkRenderer
{
    public static Renderer Create(string style)
    {
        return new Renderer
        {
            Name = "link",
            Render = (node, state, first, last, decode) => Render(node, state, style),
        };
    }

    public static void Render(MarkdownNode node, RendererState state, string style)
    {
        state.Output.Append(style);
        RenderChildren.Execute(node, state);
        if (!string.IsNullOrEmpty(node.Info))
        {
            state.Output.Append($"{RenderToConsole.AnsiReset} ({node.Info})");
        }
        else
        {
            state.Output.Append(RenderToConsole.AnsiReset);
        }
    }
}

File: swift/Sources/allmark/RenderConsole/ConsoleLinkRenderer.swift
import Foundation

@MainActor
let consoleLinkRenderer = Renderer(
	name: "link",
	render: renderConsoleLink
)

@MainActor
func renderConsoleLink(_ node: MarkdownNode, _ state: inout RendererState, _ first: Bool?, _ last: Bool?, _ decode: Bool?) {
	let styles = getConsoleStyles()
	let style = styles["link"] ?? ""
	state.output += style
	renderChildren(node: node, state: &state)
	if let info = node.info {
		state.output += "\(ansiReset) (\(info))"
	} else {
		state.output += ansiReset
	}
}
